packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const TransactionStatisticsCollector = require('../../../lib/common/core/transaction-statistics-collector');
+const TxStatus = require('../../../lib/common/core/transaction-status');
+
+const chai = require('chai');
+chai.should();
+const sinon = require('sinon');
+
+describe.only('TransactionStatisticsCollector implementation', () => {
+
+    let sandbox;
+    let clock;
+
+    beforeEach(function() {
+        sandbox = sinon.createSandbox();
+        clock = sandbox.useFakeTimers();
+    });
+
+    afterEach(function() {
+        sandbox.restore();
+    });
+
+    describe('Using a single TransactionStatisticsCollector to collect results', () => {
+
+        it('should not be possible to get basic round information from the collector', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector(0, 1, 'roundLabel');
+
+            transactionStatisticsCollector.getWorkerIndex().should.equal(0);
+            transactionStatisticsCollector.getRoundIndex().should.equal(1);
+            transactionStatisticsCollector.getRoundLabel().should.equal('roundLabel');
+        });
+
+        it('should not be possible to update a TransactionStatisticsCollector if it is not activated', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+
+            const updateSpy = sandbox.spy(transactionStatisticsCollector, '_updateStatistics');
+
+            transactionStatisticsCollector.txSubmitted(1);
+            transactionStatisticsCollector.getTotalSubmittedTx().should.equal(0);
+
+            const txStatus = new TxStatus();
+            transactionStatisticsCollector.txFinished(txStatus);
```

Comment:
Two different API methods (`txSubmitted` and `txFinished`) are tested together. Might be better to split the test into two.